src/OpenTelemetry.Instrumentation.StackExchangeRedis/Implementation/RedisProfilerEntryInstrumenter.cs

```diff
@@ -88,102 +92,148 @@ static bool GetCommandAndKey(
 
                 if (!string.IsNullOrEmpty(commandAndKey) && !string.IsNullOrEmpty(script))
                 {
-                    activity.SetTag(SemanticConventions.AttributeDbStatement, commandAndKey + " " + script);
+                    activity.SetTag(SemanticConventions.AttributeDbQueryText, commandAndKey + " " + script);
                 }
                 else if (!string.IsNullOrEmpty(commandAndKey))
                 {
-                    activity.SetTag(SemanticConventions.AttributeDbStatement, commandAndKey);
+                    activity.SetTag(SemanticConventions.AttributeDbQueryText, commandAndKey);
                 }
                 else if (command.Command != null)
                 {
-                    // Example: "db.statement": SET;
-                    activity.SetTag(SemanticConventions.AttributeDbStatement, command.Command);
+                    // Example: "db.query.text": SET;
+                    activity.SetTag(SemanticConventions.AttributeDbQueryText, command.Command);
                 }
             }
             else if (command.Command != null)
             {
-                // Example: "db.statement": SET;
-                activity.SetTag(SemanticConventions.AttributeDbStatement, command.Command);
+                // Example: "db.query.text": SET;
+                activity.SetTag(SemanticConventions.AttributeDbQueryText, command.Command);
             }
+        }
 
-            if (command.EndPoint != null)
+        if (command.EndPoint != null)
+        {
+            if (command.EndPoint is IPEndPoint ipEndPoint)
             {
-                if (command.EndPoint is IPEndPoint ipEndPoint)
-                {
-                    activity.SetTag(SemanticConventions.AttributeNetPeerIp, ipEndPoint.Address.ToString());
-                    activity.SetTag(SemanticConventions.AttributeNetPeerPort, ipEndPoint.Port);
-                }
-                else if (command.EndPoint is DnsEndPoint dnsEndPoint)
-                {
-                    activity.SetTag(SemanticConventions.AttributeNetPeerName, dnsEndPoint.Host);
-                    activity.SetTag(SemanticConventions.AttributeNetPeerPort, dnsEndPoint.Port);
-                }
-                else
-                {
-                    activity.SetTag(SemanticConventions.AttributePeerService, command.EndPoint.ToString());
-                }
+                var ip = ipEndPoint.Address.ToString();
+                var port = ipEndPoint.Port;
+
+                activity?.SetTag(SemanticConventions.AttributeServerAddress, ip);
+                activity?.SetTag(SemanticConventions.AttributeServerPort, port);
+                activity?.SetTag(SemanticConventions.AttributeNetworkPeerAddress, ip);
+                activity?.SetTag(SemanticConventions.AttributeNetworkPeerPort, port);
+
+                meterTags?.Add(SemanticConventions.AttributeServerAddress, ip);
+                meterTags?.Add(SemanticConventions.AttributeServerPort, port);
```

Comment:
ServerPort is only to be populated if it is not the default one, right?

Reply:
is there any reason not to populate it when it is the default port?

Reply:
That is what the semantic conventions states.
Also, every additional attribute [increases the cost of Metrics updates](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/test/Benchmarks/Metrics/MetricsBenchmarks.cs#L39-L45).!

Reply:
makes sense, port will be added only if not the default 6379